optax/losses/_classification.py

```diff
@@ -551,3 +552,64 @@ def sigmoid_focal_loss(
                       loss)
 
   return loss
+
+def ntxent(
+    embeddings: chex.Array,
+    labels: chex.Array,
+    temperature: float = 0.07
+) -> chex.Array:
+  """
+  Implementation of normalized temperature scaled cross entropy loss 
+  (NT-Xent).
+
+  References:
+    T. Chen, et al. 2020. http://arxiv.org/abs/2002.05709
+    kevinmusgrave.github.io/pytorch-metric-learning/losses/#ntxentloss
+
+  Args:
+    emeddings: batch of embeddings, with shape [batch, feature_length]
+    labels: labels for groups that are positive pairs. e.g. if you have
+    a batch of 4 embeddings and the first two and last two were positive
+    pairs your `labels` should look like [0, 0, 1, 1]. labels SHOULD NOT
+    be all the same (e.g. [0, 0, 0, 0]) you will get a NaN result.Shape
+    [batch]
+    temperature: temperature scaling parameter.
+
+  Returns:
+    A scalar loss value of NT-Xent values averaged over all positive
+    pairs
+  """
+  chex.assert_type([embeddings], float)
+  if labels.shape[0] != embeddings.shape[0]:
+    raise ValueError(
+      'label dimension should match batch dimension in embeddings'
+    )
+
+  eps = 1e-300
```

Comment:
Hi @vroulet! I am trying to avoid NaNs when I take the log in `log_exp = -jnp.log(softm).sum() / matches.sum()`. Is there a value that you recommend for this?

Reply:
Ah I see, normally `jax.nn.logsoftmax` handles that properly. Do you think you could code this loss without passing by exponential and only by using `jax.nn.logsoftmax` at the right time?

Reply:
I think so! I'll let you know.